.github/workflows/test-suite.yml

```diff
@@ -0,0 +1,29 @@
+---
+name: Test Suite
+
+on:
+  push:
+    branches: ["master"]
+  pull_request:
+    branches: ["master"]
+
+jobs:
+  tests:
+    name: "Python ${{ matrix.python-version }}"
+    runs-on: "ubuntu-latest"
+
+    strategy:
+      matrix:
+        python-version: ["3.6", "3.7", "3.8"]
+
+    steps:
+      - uses: "actions/checkout@v2"
+      - uses: "actions/setup-python@v1"
+        with:
+          python-version: "${{ matrix.python-version }}"
+      - name: "Install dependencies"
+        run: "scripts/install"
+      - name: "Run checks"
```

Comment:
I think it was quite nice to have the checks job separate from the tests, so that we don’t have to open the UI to see that “oh right I need to run lint”. Any chance we can replicate this on Actions?

Reply:
Very possibly, yes.

I'm not totally convinced it's necessary, but it's defo worth considering keeping that style, yes.

I figure it might be okay for us to start with the basics, and then treat anything like that as an incremental improvement, that we can consider across the whole async stack?

Reply:
Sure :)